### ISSUE TITLE
What should be the ideal memory consumption of ExoPlayer when playing a certain video?

### ISSUE THREAD

**Issue by rickyManalo** [google/ExoPlayer#6565]
### [REQUIRED] Searched documentation and issues
I Googled "ExoPlayer memory consumption/usage" already and nothing useful appeared.

### [REQUIRED] Question
I want to know what is the ideal memory usage when playing a 1.6GB video stream with a data rate of 2150kbps and a total bitrate of 2295kbps. A user said that they can't play multiple video streams on my app. I'm suspecting that it might just be a device problem. I'm trying to find if my app is creating a memory leak of some sort (not really sure, but just want to know). LeakCanary is not showing a notification so I think it's really all right. Is there like a formula I could use to get an average memory usage of ExoPlayer.

**Comment by ojw28** [comment]
If you want to see what typical memory consumption is like, I suggest you try playing something in the ExoPlayer demo app and simply attach the Android Studio profiler.

It's unclear why you're looking for memory leaks without any evidence of there being one.

**Comment by rickyManalo** [comment]
For research purposes, kinda like a bench mark. Ok, I'll try that.

**Comment by rickyManalo** [comment]
Just want to share my observation. After about 10 minutes of playing a DASH H264 720p 8000kbps using the demo app, the total memory usage got an average of 99MB. Closing PlayerActivity just made the memory usage go down to 93MB (93.1MB to be exact). 

On my app, after playing a 1080p 1.6GB video stream with a data rate of 2150kbps and a total bitrate of 2295kbps for 10 minutes as well, the total memory usage got an average of 104MB. After closing the player, the total memory usage went to an average of 54MB. On my app, I call `playerView.setPlayer(null);` when releasing the player.

I made another test on the demo app. The numbers went down by about 5MB. When I played another DASH H264 video, the total memory usage went down to 31MB then constantly increased after that.
